projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1114,9 +1114,18 @@ function is_user_auth() {
  * @return string
  */
 function get_paywall_blocks_subscribe_pending() {
-	$access_heading = esc_html__( 'Verify your email to continue reading', 'jetpack' );
+	$subscribe_email = Jetpack_Memberships::get_current_user_email();
+
+	/** This filter is documented in modules/contact-form/grunion-contact-form.php */
+	if ( is_wpcom() || false !== apply_filters( 'jetpack_auto_fill_logged_in_user', false ) ) {
```

Comment:
I think we should use the `Host` class for these checks, but we can do it in a separate PR